src/subtitle/cue.ts

```diff
@@ -0,0 +1,164 @@
+/**
```

Comment:
Where do these definitions come from? They don't resemble the iOS APIs.

Reply:
After talking to @matamegger he told me this comes from the Android project, which is using ExoPlayer types on the public API for this.

Question is what do we want to present on the public API of the React Native bridge?
A platform independent Bitmovin API that is defined by us with full flexibility or a snapshot of ExoPlayer's APIs.
I would vote for a Bitmovin API but open for further discussion.

Reply:
For correction, ExoPlayer types are not used in the API, but a design that's mimicking it to make "native" subtitle rendering easier.

As mentioned in my other comment, maybe the best is to not have positioning information for now. It's not easy to decide on how to merge 2 API surfaces into 1, that is compatible with the 2 bases directly in a PR.
I think a we would need to discuss a proposal.

Reply:
I see your point @matamegger, but I don't think that's really a hard thing to support both APIs right away. We don't need to merge them into a single API right from the beginning, even though that'd be preferable since the JS API would be as platform agnostic as possible, but at least the feature gets bridged and usable.

One alternative I pointed earlier would be break both APIs into platform-specific types: `CueiOS` and `CueAndroid`, abstract them away into a union-type, then let the user disambiguate them in runtime using TypeScript's [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) functions:

```typescript
interface CueiOS {
  kind: 'ios',
  // iOS specific props for cue type.
}

interface CueAndroid {
  kind: 'android',
  // Android specific props for cue type.
}

interface CueWeb {
  kind: 'web',
  // Web specific props for cue type.
}

// General cue type that'll be used throughout the whole API.
type Cue = CueiOS | CueAndroid | CueWeb;

// Type narrowing functions for disambiguation.
function isCueAndroid(cue: Cue): cue is CueAndroid {
  return cue.kind === 'android';
}

function isCueiOS(cue: Cue): cue is CueiOS {
  return cue.kind === 'ios';
}

function isCueWeb(cue: Cue): cue is CueWeb {
  return cue.kind === 'web';
}

// Use-case

interface CueEnterEvent {
  cue: Cue;
}

function onCueEnter(event: CueEnterEvent) {
  const { cue } = event;
  if (isCueAndroid(cue)) {
    // `cue` will be automatically of type `CueAndroid` in this scope.
  }
  if (isCueiOS(cue)) {
    // `cue` will be automatically of type `CueiOS` in this scope.
  }
  if (isCueWeb(cue)) {
    // `cue` will be automatically of type `CueWeb` in this scope.
  }
}
```

Reply:
Also, @matamegger and @rolandkakonyi, to which ExoPlayer APIs were you referring to? I have no idea of what you are talking about because I just saw @rolandkakonyi marking the first line of `cue.ts`.

I followed just what I found on the public docs. They're my only source of truth to know what I can or can't expose, so if I accidentally exposed ExoPlayer APIs, you exposed them too.

Reply:
Just ignore that ExoPlayer part. Just same API on the android player, but it still is our API so it should be considered in general.

---
> but I don't think that's really a hard thing to support both APIs right away.

I think there is different understanding of support between us. Of course it is not hard to put everything into one API. It is hard to make it meaningful and as you say platform agnostic, with the bridging to the existing APIs in the back.  What is also hard sometimes, is to get rid of API that has been introduced. Sure, we are right now in an alpha like phase, but that does not mean we should just push everything into it, just so that we can remove it later on.

What is the benefit of "supporting" this right now? I don't think that it is possible to use the RN Player without the UI components the Android and iOS SDK provide. So there will be some UI, that renders the Subtitles all the time. (Please correct me if that's not right) Which would for my understanding make (in most use-cases) custom subtitle rendering obsolete.

I honestly don't see why we need the positioning information in the cues. Also I think the base idea with `Subtitle API support` is to allow selecting and deselecting subtitle tracks from the list of available subtitle tracks, as well as adding side-loaded subtitle tracks in the configuration. Meaning the `CueEnter/Exit` events are actually not relevant (at the moment). You also do not mention the `Cue` in the ticket: https://github.com/bitmovin/bitmovin-player-react-native/issues/7
I am not aware of scope changes, so if the information I have is outdate please let me know.

Reply:
Ok, I see your point. Initially, my intention was to try to bridge the subtitles API as much as I could, including cue events, but since the scope of what must be bridged from cues is still blurry, I'll drop them. It's perfectly fine.

 And you're right. Even though it would be nice to have some APIs supported from the beginning, making people dependent on a poor interface will just make it harder to drop it later. But I disagree about custom subtitles rendering getting obsolete in most use-cases, it seems to be a common issue for people trying to use Bitmovin player with RN ([see](https://stackoverflow.com/questions/70196742/how-do-i-layout-my-react-native-fire-tv-subtitleview-correctly)). So I still think it should be an important part of RN's implementation.

But anyways, I think it's fine for us to discuss this in more detail once it gets prioritized later. Thanks 🙂.

Reply:
Following up this discussion, cues support has been dropped for now: 1599353362e30aa2471b60d7ddcff02d88bf8b2c